chroma-agent/chroma_agent/device_plugins/linux.py

```diff
@@ -22,84 +14,9 @@ class LinuxDevicePlugin(DevicePlugin):
-        # solution for HYD-3140. This plugin should really be loaded if it is not needed but for now this sorts out
-        # and issue with PluginAgentResources being in the linux plugin.
-        if config.get('settings', 'profile')['worker']:
-            return {}
-
-        # Before we do anything do a partprobe, this will ensure that everything gets an up to date view of the
-        # device partitions. partprobe might throw errors so ignore return value
-        AgentShell.run(["partprobe"])
-
-        # Map of block devices major:minors to /dev/ path.
-        block_devices = BlockDevices()
-
-        # Devicemapper: LVM and Multipath
-        dmsetup = DmsetupTable(block_devices)
-
-        # Software RAID
-        mds = MdRaid(block_devices).all()
-
-        # _zpools
-        zfs_devices = ZfsDevices()
-        zfs_devices.full_scan(block_devices)
-
-        # EMCPower Devices
-        emcpowers = EMCPower(block_devices).all()
-
-        # Local filesystems (not lustre) in /etc/fstab or /proc/mounts
-        local_fs = LocalFilesystems(block_devices).all()
-
-        # We have scan devices, so set the devices scanned flags.
-        LinuxDevicePlugin.devices_scanned = True
-
-        return {"vgs": dmsetup.vgs,
-                "lvs": dmsetup.lvs,
-                "zfspools": zfs_devices.zpools,
-                "zfsdatasets": zfs_devices.datasets,
-                "zfsvols": zfs_devices.zvols,
-                "mpath": dmsetup.mpaths,
-                "devs": block_devices.block_device_nodes,
-                "local_fs": local_fs,
-                'emcpower': emcpowers,
-                'mds': mds}
-
-    def _scan_devices(self, scan_always):
-        full_scan_result = None
-
-        if scan_always or (self._quick_scan() != self._last_quick_scan_result):
-            self._last_quick_scan_result = self._quick_scan()
-            full_scan_result = self._full_scan()
-            self._last_full_scan_result = full_scan_result
-        elif self._safety_send < DevicePlugin.FAILSAFEDUPDATE:
-            self._safety_send += 1
-        else:
-            # The purpose of this is to cause the ResourceManager to re-evaluate the device-graph for this
-            # host which may lead to different results if the devices reported from other hosts has changed
-            # This should not really be required but is a harmless work around while we get the manager code
-            # in order
-            full_scan_result = self._last_full_scan_result
-
-        if full_scan_result is not None:
-            self._safety_send = 0
-
-        return full_scan_result
 
     def start_session(self):
-        return self._scan_devices(True)
+        return {}
 
     def update_session(self):
-        trigger_plugin_update = self.trigger_plugin_update
-        self.trigger_plugin_update = False
-        scan_result = self._scan_devices(trigger_plugin_update)
-
-        return scan_result
+        return {}
```

Comment:
given this is now basically a no-op, any reason we can't drop the whole plugin?

Reply:
we still currently use the plug-in mechanism to trigger manager side code, that will be removed later